### PR TITLE
Add simplified H2 feedin restriction

### LINKED ISSUE #790: Add simplified variant for H2 feed-in restrictions into CH4 grid
The H2 feed-in restrictions will be simplified within the eGon-data workflow by assigning a p_nom value, that is a fraction of the CH4 transport capacity of the connected pipelines. The modeling of H2 feed-in restrictions proposed in eTraGo (see https://github.com/openego/eTraGo/pull/392) are quite complex and there should be a more simple approach available.

### PR DESCRIPTION
**Pull request by fwitte**
Fixes #790

## Before merging into `dev`-branch, please make sure that

- [x] the `CHANGELOG.rst` was updated.
- [x] new and adjusted code is formated using `black` and `isort`.
- [x] the `Dataset`-version is updated when existing datasets are adjusted.
- [x] the branch was merged into the
      `continuous-integration/run-everything-over-the-weekend`-branch.
- [x] the workflow is running successful in `test mode`.
- [x] the workflow is running successful in `Everything` mode.

### CODE REVIEW

File: src/egon/data/datasets/hydrogen_etrago/h2_to_ch4.py [hunk 1]
```diff
@@ -88,3 +121,68 @@ def insert_h2_to_ch4_eGon100RE():
...
+    capacity (based on energy) therefore decreases if the volumetric flow
+    does not change. This effect is neglected in eGon. At 15 vol% H2 the
+    decrease in capacity equals about 10 % if volumetric flow does not change.
+
+    Parameters
+    ----------
+    x : float
+        Volumetric percentage of H2 in the mixture
+    T : int, optional
+        Temperature of the mixture, by default 25
```
Comment (AmeliaNadal) [inline src/egon/data/datasets/hydrogen_etrago/h2_to_ch4.py:143]: You could mention the unit of T

File: src/egon/data/datasets/hydrogen_etrago/h2_to_ch4.py [hunk 2]
```diff
@@ -88,3 +121,68 @@ def insert_h2_to_ch4_eGon100RE():
...
+    p : int, optional
+        Pressure of the mixture, by default 50
```
Comment (AmeliaNadal) [inline src/egon/data/datasets/hydrogen_etrago/h2_to_ch4.py:145]: Same, unit could be mentionned

**Review by AmeliaNadal — APPROVED** [review]

File: src/egon/data/datasets/hydrogen_etrago/h2_to_ch4.py (hunk 1 above)
Comment (fwitte) [reply src/egon/data/datasets/hydrogen_etrago/h2_to_ch4.py]: added

File: src/egon/data/datasets/hydrogen_etrago/h2_to_ch4.py (hunk 2 above)
Comment (fwitte) [reply src/egon/data/datasets/hydrogen_etrago/h2_to_ch4.py]: added

**Pull request merged**